src/test/java/com/kodilla/ecommercee/domainTests/ProductTestSuite.java

```diff
@@ -0,0 +1,162 @@
+    public void cleanup() {
+        productRepository.deleteAll();
+        groupRepository.deleteAll();
+    }
+
+    @Test
+    public void testAddProductToDatabase() {
+    // Given
+        Group group = new Group();
+        group.setName("Test group");
+        group.setDescription("Test Group Description");
+        Group savedGroup = groupRepository.save(group);
+        Product product = new Product();
+        product.setName("Test Product");
+        product.setDescription("Test Description");
+        product.setPrice(BigDecimal.valueOf(10.00));
+        product.setGroup(savedGroup);
+        product.setCarts(Arrays.asList(new Cart()));
+        product.setOrders(Arrays.asList(new Order()));
+        product.setActive(true);
+
+    //When
+     Product savedProduct = productRepository.save(product);
+    //Then
+        assertNotNull(savedGroup);
+        assertEquals("Test Product", savedProduct.getName());
+        assertEquals("Test Description", savedProduct.getDescription());
+        assertEquals(BigDecimal.valueOf(10.00), savedProduct.getPrice());
+        assertNotNull(savedProduct.getGroup());
+        assertEquals(1, savedProduct.getCarts().size());
+        assertEquals(1, savedProduct.getOrders().size());
+        assertTrue(savedProduct.isActive());
+    //CleanUp
+        productRepository.deleteAll();
+        groupRepository.deleteAll();
+    // Verify cleanup
+        assertEquals(0, productRepository.findAll().size());
+        assertEquals(0, groupRepository.findAll().size());}
+    @Test
+    public void testFindProductById() {
+    // Given
+        Product product = new Product();
+        product.setName("Test Product");
+        product.setDescription("Test Description");
+        product.setPrice(BigDecimal.valueOf(10.00));
+        product.setActive(true);
+        Product savedProduct = productRepository.save(product);
+
+    // When
+        Long productId = savedProduct.getId();
+        Product retrievedProduct = productRepository.findById(productId).orElse(null);
+
+    // Then
+        assertNotNull(retrievedProduct);
+        assertEquals(productId, retrievedProduct.getId());
+        assertEquals("Test Product", retrievedProduct.getName());
+        assertEquals("Test Description", retrievedProduct.getDescription());
+        assertEquals(BigDecimal.valueOf(10.00), retrievedProduct.getPrice());
+        assertTrue(retrievedProduct.isActive());
+    }
+    @Test
+    public void testDeleteProduct() {
+        // Given
+        Product product = new Product();
+        product.setName("Test Product");
+        product.setDescription("Test Description");
+        product.setPrice(BigDecimal.valueOf(10.00));
+        product.setActive(true);
+        Product savedProduct = productRepository.save(product);
+
+        // When
+        productRepository.delete(savedProduct);
+
+        // Then
+        Optional<Product> deletedProductOptional = productRepository.findById(savedProduct.getId());
```

Comment:
tutaj podobnie mi się wydaje jak w teście testFindProductById(), ale ktoś niech po mnie jeszcze sprawdzi

```suggestion
                productRepository.save(product);

        // When
        productRepository.delete(product);

        // Then
        Optional<Product> deletedProductOptional = productRepository.findById(product.getId());
```